wandb/sdk/internal/internal_api.py

```diff
@@ -141,7 +141,11 @@ def __init__(
             load_settings=load_settings,
             root_dir=self.default_settings.get("root_dir"),
         )
-        self.git = GitRepo(remote=self.settings("git_remote"))
+        # TODO: ensure wandb.Settings is getting passed in from setup or wandb.init
+        self.git = GitRepo(
+            remote=self.settings("git_remote"),
+            disabled=bool(self.default_settings.get("disable_git", False)),
```

Comment:
Ah all of these setting should be unified to the new`wandb.Setting` :/

Reply:
probably should add `disable_git` to `DefaultSettings`? I think in that case you won't need to do the type casting...